### ISSUE TITLE
Hexadecimal constants bug

### ISSUE THREAD

**Issue by fawltylanguage** [gnudatalanguage/gdl#1086]
```
IDL> help, '8000000000000000'xll
<Expression>    LONG64    =   -9223372036854775808
IDL> help, '8fffffffff000000'xll
<Expression>    LONG64    =   -8070450532264706048
IDL>

GDL> help, '8000000000000000'xll
<Expression>    LONG64    =                     -1
GDL> help, '8fffffffff000000'xll
<Expression>    LONG64    =                     -1
GDL>
```
If the sign bit is set, the result is always -1.

**Comment by GillesDuvert** [comment]
GDL checks an 'overflow' in Text2Number() (file dnode.cpp)
If overflow is not checked in the xll particular case, the returned value is OK.
So the 'overflow' computation for LLs is faulty.

**Comment by GillesDuvert** [comment]
additionnaly:
```
IDL> help, '8000000000000000'x
<Expression>    LONG64    =   -9223372036854775808
GDL> help, '8000000000000000'x  
<Expression>    INT       =        0
```

**Comment by GillesDuvert** [comment]
solved in #1204
